bonding-farm/contracts/AutoStakingBond.sol

```diff
@@ -0,0 +1,888 @@
+import '@openzeppelin/contracts/token/ERC721/extensions/IERC721Enumerable.sol';
+
+import './interfaces/IPriceOracleAggregator.sol';
+
+interface IOwnableUpgradeable {
+    function policy() external view returns (address);
+
+    function renounceManagement() external;
+
+    function pushManagement(address newOwner_) external;
+
+    function pullManagement() external;
+}
+
+abstract contract OwnableUpgradeable is
+    IOwnableUpgradeable,
+    Initializable,
+    ContextUpgradeable
+{
+    address internal _owner;
+    address internal _newOwner;
+
+    event OwnershipPushed(
+        address indexed previousOwner,
+        address indexed newOwner
+    );
+    event OwnershipPulled(
+        address indexed previousOwner,
+        address indexed newOwner
+    );
+
+    /**
+     * @dev Initializes the contract setting the deployer as the initial owner.
+     */
+    function __Ownable_init() internal initializer {
+        __Context_init_unchained();
+        __Ownable_init_unchained();
+    }
+
+    function __Ownable_init_unchained() internal initializer {
+        _owner = msg.sender;
+        emit OwnershipPushed(address(0), _owner);
+    }
+
+    function policy() public view override returns (address) {
+        return _owner;
+    }
+
+    modifier onlyPolicy() {
+        require(_owner == msg.sender, 'Ownable: caller is not the owner');
+        _;
+    }
+
+    function renounceManagement() public virtual override onlyPolicy {
+        emit OwnershipPushed(_owner, address(0));
+        _owner = address(0);
+    }
+
+    function pushManagement(address newOwner_)
+        public
+        virtual
+        override
+        onlyPolicy
+    {
+        require(
+            newOwner_ != address(0),
+            'Ownable: new owner is the zero address'
+        );
+        emit OwnershipPushed(_owner, newOwner_);
+        _newOwner = newOwner_;
+    }
+
+    function pullManagement() public virtual override {
+        require(msg.sender == _newOwner, 'Ownable: must be new owner to pull');
+        emit OwnershipPulled(_owner, _newOwner);
+        _owner = _newOwner;
+    }
+}
+
+library CountersUpgradeable {
+    struct Counter {
+        uint256 _value; // default: 0
+    }
+
+    function init(Counter storage counter, uint256 _initValue) internal {
+        counter._value = _initValue;
+    }
+
+    function current(Counter storage counter) internal view returns (uint256) {
+        return counter._value;
+    }
+
+    function increment(Counter storage counter) internal {
+        counter._value += 1;
+    }
+
+    function decrement(Counter storage counter) internal {
+        counter._value = counter._value - 1;
+    }
+}
+
+interface ILockFarm {
+    function stake(uint256 amount, uint256 secs) external;
+
+    function withdraw(uint256 fnftId) external;
+
+    function rewardToken() external view returns (address);
+}
+
+contract AutoStakingBond is OwnableUpgradeable, PausableUpgradeable {
```

Comment:
@bejaxer since AutoStaking is just one of the features of the bond v3, I wouldn't name the bond contract with "Autostaking". 

Will you name the contract name something like "BondNoTreasury" and update the version to 3.1?